### DISCUSSION TITLE
Many errors when running the service and no name resolution on a QNAP NAS

### DISCUSSION THREAD

**Discussion by Xandr3** [qdm12/ddns-updater#272] [Q&A] [answered]
- [x] Latest version of ddns-updater (v2.4.1)
- [x] DNS provider : OVH
- [x] Check Issues and Discussions

Hello,

I recently installed DDNS on 2 Qnap NAS from a package created for.

My 2 NAS are similar in terms of configuration and OS. It works perfectly on one of them only.

On the NAS where it doesn't work, I have the following errors:
```
[/opt/ddnsupdater] # ./ddnsupdater.sh start
[/opt/ddnsupdater] # =========================================
=========================================
========= DDNS Universal Updater ========
=========================================
=== Made with ❤️ by github.com/qdm12 ====
=========================================

Running version unknown built on an unknown date (commit unknown)


🔧 Need help? https://github.com/qdm12/ddns-updater/issues/new
💻 Email? quentin.mcgaw@gmail.com
☕ Slack? Join from the Slack button on Github
💸 Help me? https://github.com/sponsors/qdm12
2021/11/28 15:59:43 INFO Found single setting to update record
2021/11/28 15:59:43 WARN failed connectivity check: for github.com: lookup github.com: no such host
2021/11/28 15:59:43 WARN failed connectivity check: for github.com: Get "https://github.com": dial tcp: lookup github.com: no such host
2021/11/28 15:59:43 WARN failed connectivity check: for github.com: Get "http://github.com": dial tcp: lookup github.com: no such host
2021/11/28 15:59:43 INFO Reading history from database: domain thor.gradarom.net host @
2021/11/28 15:59:43 INFO healthcheck server: listening on 127.0.0.1:9999
2021/11/28 15:59:43 ERROR Get "https://api.ipify.org": dial tcp: lookup api.ipify.org: no such host
2021/11/28 15:59:43 INFO backup: disabled
2021/11/28 15:59:43 INFO http server: listening on :26000
2021/11/28 15:59:43 WARN cannot DNS resolve my.personnaldns.net after 5 tries: lookup my.personnaldns.net: no such host
```

I checked the "updates.json" file, but it does not contain any IP address.

If I restart the application, I get the following error (because I have no IP address).
```
[/opt/ddnsupdater] # =========================================
=========================================
========= DDNS Universal Updater ========
=========================================
=== Made with ❤️ by github.com/qdm12 ====
=========================================

Running version unknown built on an unknown date (commit unknown)


🔧 Need help? https://github.com/qdm12/ddns-updater/issues/new
💻 Email? quentin.mcgaw@gmail.com
☕ Slack? Join from the Slack button on Github
💸 Help me? https://github.com/sponsors/qdm12
2021/11/28 16:02:27 ERROR /share/MD0_DATA/.qpkg/ddnsupdater/data/updates.json validation error: IP 1 of 1 is empty for record {"domain": "my.personnaldns.net", "host":"@", "ips":[{"ip":"", "time": "2021-11-28T16:02:04.440650328+01:00"}]}
2021/11/28 16:02:27 INFO Shutdown successful
```

I tried a DNS configuration crossover between the two NAS and it's not an issue related to that.

Nslookup works perfectly :
```
Name:       github.com
Address:    140.82.121.4
```

I have contacted the person who packaged the application for the QNAPs, but have not had a response.

Would you have any information to solve my problem?

Thanks ;)

**Comment by qdm12** [comment]
Do you have a firewall you run for your containers? It looks like it doesn't have DNS access.

Otherwise you can try setting `--dns 1.1.1.1` in your docker run commnd.

**Reply by Xandr3** [reply]
Hi,
I don't use docker, but this : https://www.qnapclub.eu/fr/qpkg/1190

I don't have a firewall out of my network.

The "nslookup" command resolves the request to github well (as shown above).

The command below shows the DNS configuration of my NAS: 
```
[~] # cat /etc/resolv.conf
nameserver 192.168.1.1
nameserver 8.8.8.8
```

The first address being my ISP's modem. The DNS configuration is identical to the one on the NAS which works.

I can't force the "dns" when running the program. Can I specify this in the configuration file?

**Reply by Xandr3** [reply]
Hi,

Can I specify "dns 1.1.1.1" in the config.json file ? If yes, under which syntax?

**Reply by xandau** [reply]
hi,

previously my dns didnt resolve properly, due to that the ip was left empty in the update.json.

If you can resolve the domain like in the example shown by @qdm12 in Issue #119  sucsessfully then removing the update.json file  and restarting the service solved my problem.

**Reply by qdm12** [reply, marked as answer]
On the latest image, you can now use `RESOLVER_ADDRESS` to set a DNS address 😉 For example `RESOLVER_ADDRESS=1.1.1.1`.

> previously my dns didnt resolve properly, due to that the ip was left empty in the update.json.

This is a bug being investigated, it should be resolved in the coming week.
